pkg/apis/core/v1beta1/validation/validation.go

```diff
@@ -139,6 +143,71 @@ func ValidateFederatedTypeConfigStatus(status *v1beta1.FederatedTypeConfigStatus
 	return allErrs
 }
 
+func ValidateKubeFedConfig(kubeFedConfig *v1beta1.KubeFedConfig) field.ErrorList {
+	klog.V(2).Infof("Validating KubeFedConfig %q", kubeFedConfig.Name)
+
+	allErrs := field.ErrorList{}
+
+	spec := kubeFedConfig.Spec
+	specPath := field.NewPath("spec")
+	allErrs = append(allErrs, validateEnumStrings(specPath.Child("scope"), string(spec.Scope),
+		[]string{string(apiextv1b1.ClusterScoped), string(apiextv1b1.NamespaceScoped)})...)
+
+	duration := spec.ControllerDuration
+	durationPath := specPath.Child("controllerDuration")
+	allErrs = append(allErrs, validateGreaterThan0(durationPath.Child("availableDelay"), int64(duration.AvailableDelay.Duration))...)
+	allErrs = append(allErrs, validateGreaterThan0(durationPath.Child("unavailableDelay"), int64(duration.UnavailableDelay.Duration))...)
+
+	elect := spec.LeaderElect
+	electPath := specPath.Child("leaderElect")
+	allErrs = append(allErrs, validateGreaterThan0(electPath.Child("leaseDuration"), int64(elect.LeaseDuration.Duration))...)
+	allErrs = append(allErrs, validateGreaterThan0(electPath.Child("renewDeadline"), int64(elect.RenewDeadline.Duration))...)
+	allErrs = append(allErrs, validateGreaterThan0(electPath.Child("retryPeriod"), int64(elect.RetryPeriod.Duration))...)
+	if elect.LeaseDuration.Duration <= elect.RenewDeadline.Duration {
+		allErrs = append(allErrs, field.Invalid(electPath.Child("leaseDuration"), elect.LeaseDuration,
+			"leaseDuration must be greater than renewDeadline"))
+	}
+	if elect.RenewDeadline.Duration <= time.Duration(float64(elect.RetryPeriod.Duration)*leaderelection.JitterFactor) {
+		allErrs = append(allErrs, field.Invalid(electPath.Child("renewDeadline"), elect.RenewDeadline,
+			"renewDeadline must be greater than retryPeriod*JitterFactor"))
```

Comment:
Please include the jitter factor in the message.  

Also, is there a reason the jitter factor isn't configurable?

cc: @shashidharatd

Reply:
Seems `JitterFactor` is agreed upon constant https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go#L72 and is not exposed by the leader election library. lets keep it like that for now.

Reply:
Ok, so it should remain non-configurable.  Is there a reason it should not be exposed in the error message though?  I think the value should either be explicitly mentioned in the error message or mention of `JitterFactor` removed.